smart_contract/executor/derive/src/validate.rs

```diff
@@ -41,7 +41,6 @@ impl FromAttributes for ValidateAttribute {
     // but we still _want_ to validate that each attribute parses successfully
     // this is to ensure that we provide the user with as much validation as possible, instead of bailing out early
     // `Option::or_else` would NOT work here, as it would not validate conditions after the first valid one
-    #[allow(clippy::or_fun_call)]
```

Comment:
Please see the comment above, this is NOT correct, this code actually DOES rely on the `or` argument to be executed every time